ballista/rust/executor/src/flight_service.rs

```diff
@@ -199,15 +189,21 @@ fn create_flight_iter(
     )
 }
 
-async fn stream_flight_data<T>(
-    reader: FileReader<T>,
-    tx: FlightDataSender,
-) -> Result<(), Status>
-where
-    T: Read + Seek,
-{
-    let options = arrow::ipc::writer::IpcWriteOptions::default();
-    let schema_flight_data = SchemaAsIpc::new(reader.schema().as_ref(), &options).into();
+async fn stream_flight_data(path: String, tx: FlightDataSender) -> Result<(), Status> {
+    let mut file = File::open(&path)
+        .map_err(|e| {
+            BallistaError::General(format!(
+                "Failed to open partition file at {}: {:?}",
+                path, e
+            ))
+        })
+        .map_err(|e| from_ballista_err(&e))?;
+    let file_meta = read_file_metadata(&mut file).map_err(|e| from_arrow_err(&e))?;
+    let reader = FileReader::new(&mut file, file_meta, None);
```

Comment:
Move file open as well as FileReader creation logic here for the lifetime requirements enforced by async move {...}